### PR TITLE
Fix line wrapping for Internal Links

### PR DESCRIPTION
**Pull request by santdeleon**
## Description

- Added css styling to internal links to remove line wrapping. 
- Broke the **page-eth2-staking-validators-desc** json value out into three parts/components such that one of the components can become a glossary link to the 51% attack url (**/glossary/#51-attack**).

**Eth2 Staking Validators description changes**
- The json value pair for **"page-eth2-staking-validators-desc"** has become three pairs.
1) **page-eth2-staking-validators-desc**
2) **page-eth2-staking-validators-desc-link**
3) **page-eth2-staking-validators-desc-2**
This allows us to break out and turn one portion of the description into a link.

**Link Changes**
- The `<InternalLink />` component was given the css style `white-space: nowrap;`. This keeps `<GlossaryIcon />` components from wrapping onto new lines when a user's screen is resized. It also keeps the entirety of all internal links from line wrapping. In short, internal links will always show the full length of the link at any given time.
## Related Issue

https://github.com/ethereum/ethereum-org-website/issues/1921

### CODE REVIEW

File: src/components/Link.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ const ExternalLink = styled.a`
 `
 
 const InternalLink = styled(IntlLink)`
+  white-space: nowrap;
```
Comment (wackerow) [inline src/components/Link.js:34]: Unfortunately this will affect all other internal links, and force them to not wrap at all when they should word-wrap.

**Review by wackerow — CHANGES_REQUESTED** [review]
Hey @santdeleon thanks for taking a look at this! This would potentially fix that specific link commented on in the original issue, but changing `InternalLink` to `nowrap` will affect many other links, and ultimately these should word-wrap, with the issue being just the `[?]` icon that trails `/glossary/` links specifically should stick to the last word.

There has also been a push recently to limit the breaks within sentences that require translation, as this complicates releasing these pages in other languages with proper syntax.  Hope this helps add some guidance on what the goal is here, and I certainly appreciate the work. 

Btw- Took a look at this issue myself previously and it's tricky since the way it works on the 'External Links' is it uses a css pseudo-element `::after` with `content: "↗";` to add the arrow, which is text. In this situation we use an `Icon` which is a React component, and that cannot be passed into the css `content` property like `↗` can. Another attempt was to play with margin-right of the text for glossary links to make room for the Icon on the right, but this also does not work properly with word-wrapping on inline items. 🤷‍♂️

**Comment by santdeleon** [comment]
@wackerow I see. I thought that might be a problem. This is a bit tricky but I think I have a better solution.

What about replacing the `whitespace: nowrap` with 
```
white-space: ${({ className }) =>
    className && className.includes("nowrap") && "nowrap"};
```
With this solution all you would need to do is pass a `"nowrap"` className to your Link component and then the css can conditionally render the apply `white-space` styling as necessary.

I'm sure there are many ways to solve this but this might have the least overhead?

I have it working locally and am happy to send it through if you think this could work.

**Comment by wackerow** [comment]
Right, this would help make it more selective, but my concern would be that ultimately the `nowrap` value causes the entire inline item (the entire link text) to stay on one line, versus getting only the trailing `[?]` icon to "not wrap." I previously tried applying `nowrap` to just the glossary icon but it doesn't work that way since it's a whole new inline item. 

Ultimately not up to me though, feel free to push changes! =)

**Comment by santdeleon** [comment]
Right right. Hmm. I think this may be our best bet at the moment. The only way I can think of that works for sure is using `content` like we do with ExternalLink, but since content can't take html/jsx we can't render the icon. 😣

I'll push my current fix for now and touch back on this pending team approval etc.

**Review by samajammin — CHANGES_REQUESTED** [review]
Thanks for looking into this @santdeleon! Thank you @wackerow for the assist as well 😄 

@wackerow brought up a valid point - we've made a push to remove these type of fragmented links because they break up sentences, making them difficult to translate (more context in #2047 & #2104). As a result your changes to the Staking page are outdated & no longer necessary.

As for the styling changes, I think your solution is not ideal (since it forces the entire link to `nowrap`) BUT I think it's better than the current situation. However I believe there's a way to accomplish this without requiring the explicit class name for glossary links. Since we already have a `isGlossary` variable in the Link component, let's just pass that boolean as a prop to our `InternalLink` component. Then you can add your logic based on that, e.g. something like:
```
const InternalLink = styled(IntlLink)`
  white-space: ${(props) => (props.isGlossary ? `nowrap` : `normal`)};
  ...
```

Please make that update & remove the changes the Staking files & we can get this merged. Thanks again!

**Comment by santdeleon** [comment]
Hey @samajammin! Upon your suggestions above, it seems this PR becomes invalid after implementing them.

So when I revert the staking page and wrap the description in a `Link` component (which I assume is what was expected) the whole description becomes a link. Then applying the style changes using the `isGlossary` prop like in the snippet you left above, we end up with one giant link that sticks to one line, never wraps, and overflows on the X-axis. 

I think my solution only works (at least for this specific issue) if the descriptions are broken up.

**So i'm wondering:**
1. If the changes in this PR to need to be abandoned entirely
2. Or if we actually just intend to remove the `Link` component entirely and leave the description "link-less" (like how it is in prod) - making the style changes unnecessary.

Sorry If I explained that in a confusing way! Please do let me know your thoughts. Thanks!

**Comment by samajammin** [comment]
@santdeleon hmmm... not sure I follow entirely. I wouldn't worry too much about the specific example in the the issue. I think merely making that style change I mentioned above (& only that change) would suffice here. LMK if you have any other thoughts / questions. Thanks again!

**Review by samajammin — APPROVED** [review]
Thanks again!

**Comment by samajammin** [comment]
@all-contributors please add @santdeleon for code.

**Pull request merged**